pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -673,6 +673,13 @@ type Module struct {
 	// Enabled determines whether a module should be enabled or not
 	// +optional
 	Enabled bool `json:"enabled,omitempty"`
+	// Settings to further configure the module
+	Settings ModuleSettings `json:"settings,omitempty"`
+}
+
+type ModuleSettings struct {
+	// BalancerMode sets the `balancer` module with different modes like `upmap`, `crush-compact` etc
+	BalancerMode string `json:"balancerMode,omitempty"`
```

Comment:
How about adding a `+kubebuilder:validation:Enum` for the possible values?

Reply:
Added enums for currently available modes 
` // +kubebuilder:validation:Enum="";crush-compat;upmap`

Reply:
Another mode I understand coming soon is `upmap-read`, though not sure the exact ceph version. Maybe we just add it to the enum now?

Reply:
It will be a concern if we add it to the enum while not supporting it.   Other option would be to add it to the enum and also add a ceph version check in the operator code, that is, fail with error if `upmap-read` is selected with unsupported version.

Reply:
What error do we get from ceph if Rook attempts to apply an invalid option? As long as it fails the reconcile, IMO that will be sufficient, and Rook wouldn't need an extra ceph version check unless we want to wrap it with a better error message.

Reply:
The flow of configuring the modules is a[ go routine](https://github.com/rook/rook/blob/53be5ce1e928674a5e1866dcb68370cdfba17d61/pkg/operator/ceph/cluster/mgr/mgr.go#L396).  It retires for a few times to apply then invalid option, then reconciles.  Since its a goroutine, the creation the OSD daemon is not stopped in case there is an error. 

``` 
2024-05-27 07:00:09.020100 E | op-mgr: failed modules: "mgr module(s) from the spec". failed to configure module "balancer": failed to set balancer module mode to "upmap-read": failed to set mgr module mode "upmap-read" even after 5 retries: failed to set balancer mode "upmap-read": exit status 22
```

Reply:
Will `upmap-read` be available in Squid? If so, we just as well add it to the schema now since Squid will be out soon. If they use it on an unsupported version, this seems like an ok error.

Reply:
The email conversation with Neha mentions that it was already merged with Squid.  Added it to the enum now.